File: Hash Table/350. Intersection of Two Arrays II/intersection.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        set1 = [0] * 1001
        set2 = [0] * 1001
        intersect = []

        for item in nums1:
            set1[item] += 1

        for item in nums2:
            set2[item] += 1

        for i in range(1001):
            count = min(set1[i], set2[i])
            while count:
                intersect.append(i)
                count -= 1

        return intersect

File: Hash Table/350. Intersection of Two Arrays II/intersection.cpp
class Solution
{
public:
    vector<int> intersect(vector<int> &nums1, vector<int> &nums2)
    {
        vector<int> set1(1001, 0);
        vector<int> set2(1001, 0);
        vector<int> intersect;

        for (auto item : nums1)
        {
            set1[item] += 1;
        }

        for (auto item : nums2)
        {
            set2[item] += 1;
        }

        for (int i = 0; i < 1001; i++)
        {
            int count = min(set1[i], set2[i]);

            while (count--)
            {
                intersect.push_back(i);
            }
        }
        return intersect;
    }
};